src/common/base/test/RangeVsTransformBenchmark.cpp

```diff
@@ -0,0 +1,130 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+#include <folly/init/Init.h>
+#include <folly/Benchmark.h>
+#include <algorithm>
+#include <vector>
+
+
+BENCHMARK(Test1_RangeTestStr) {
+    std::vector<int32_t> from;
+    std::vector<std::string> to;
+    BENCHMARK_SUSPEND {
+        from.resize(1000, 0);
+        to.resize(1000);
+    }
+    int32_t index = 0;
+    for (auto& i : from) {
+        to[index++] = std::to_string(i);
+    }
+    folly::doNotOptimizeAway(to);
+}
+BENCHMARK_RELATIVE(Test1_TransformStr) {
+    std::vector<int32_t> from;
+    std::vector<std::string> to;
+    BENCHMARK_SUSPEND {
+        from.resize(1000, 0);
+        to.resize(1000);
+    }
+    std::transform(from.begin(), from.end(), to.begin(), [] (const auto& e) {
+        return std::to_string(e);
+    });
+    folly::doNotOptimizeAway(to);
+}
+
+BENCHMARK_DRAW_LINE();
+
+BENCHMARK(Test2_RangeTestInt) {
+    std::vector<int32_t> from;
+    std::vector<int32_t> to;
+    BENCHMARK_SUSPEND {
+        from.resize(1000, 0);
+        to.resize(1000);
+    }
+    int32_t index = 0;
+    for (auto& i : from) {
+        to[index++] = i;
+    }
+    folly::doNotOptimizeAway(to);
+}
+BENCHMARK_RELATIVE(Test2_TransformInt) {
+    std::vector<int32_t> from;
+    std::vector<int32_t> to;
+    BENCHMARK_SUSPEND {
+        from.resize(1000, 0);
+        to.resize(1000);
+    }
+    std::transform(from.begin(), from.end(), to.begin(), [] (const auto& e) {
+        return e;
+    });
+    folly::doNotOptimizeAway(to);
+}
+
+BENCHMARK_DRAW_LINE();
+
+BENCHMARK(Test3_RangeTestInt) {
+    std::vector<int32_t> v;
+    BENCHMARK_SUSPEND {
+        v.resize(1000, 0);
+    }
+    for (auto& i : v) {
+        (void)(i);
+        int a;
+        folly::doNotOptimizeAway(a);
+    }
+}
+
+BENCHMARK_RELATIVE(Test3_ForEachInt) {
+    std::vector<int32_t> v;
+    BENCHMARK_SUSPEND {
+        v.resize(1000, 0);
+    }
+    std::for_each(v.begin(), v.end(), [] (const auto&) {
+        int a;
+        folly::doNotOptimizeAway(a);
+    });
+}
+
+int main(int argc, char** argv) {
+    folly::init(&argc, &argv, true);
+    folly::runBenchmarks();
+    return 0;
+}
+
+
+/*
+-O2
+[heng.chen@hp-server nebula]$ src/common/base/test/_build/range_vs_transform__bm
```

Comment:
Better no to expose such kind of privacy.

Better post the CPU model you are testing on.